src/OSPSuite.SimModelNative/src/Species.cpp

```diff
@@ -37,6 +37,45 @@ Species::~Species(void)
 		delete[] _RHS_UsedVariablesIndices;
 		_RHS_UsedVariablesIndices = NULL;
 	}
+	for (const auto& myPair : _jacobian_parameter_map) {
+		delete myPair.second;
+	}
+	for (const auto& myPair : _jacobian_state_variable_map) {
+		delete myPair.second;
+	}
+}
+
+Formula* calculateJacobianFor(const int index, Species* species)
+{
+	const auto& formula = species->DE_Jacobian(index);
+	return formula->RecursiveSimplify();
+}
+
+void Species::CalculateJacobianParameterFor(const int parameterIndex) 
```

Comment:
I would rename it to something like `CacheJacobianFormulaForParameter `or `CreateJacobianFormulaForParameter`.
Same for `calculateJacobianFor `and `CalculateJacobianStateVariableFor`
When I see a `CalculateXY `function - I expect that it returns a (numerical) value.